dev/breeze/src/airflow_breeze/commands/sbom_commands.py

```diff
@@ -291,25 +291,74 @@ def generate_providers_requirements(
         with open(PROVIDER_METADATA_JSON_FILE_PATH) as f:
             provider_metadata = json.load(f)
             providers_info = [
-                (p_id, p_version, info["associated_airflow_version"])
-                for (p_id, p_versions) in provider_metadata.items()
-                for (p_version, info) in p_versions.items()
+                (
+                    provider_id,
+                    provider_version,
+                    python_version,
+                    info["associated_airflow_version"],
+                )
+                for (provider_id, provider_versions) in provider_metadata.items()
+                for (provider_version, info) in provider_versions.items()
+                for python_version in AIRFLOW_PYTHON_COMPATIBILITY_MATRIX[info["associated_airflow_version"]]
+                if python_version in python_versions
             ]
     else:
         if airflow_version is None:
             airflow_version = get_active_airflow_versions(confirm=False)[-1]
             get_console().print(f"[info]Using {airflow_version} as airflow version")
-        providers_info = [(provider_id, None, airflow_version) for provider_id in provider_ids]
+        providers_info = [
+            (provider_id, None, python_version, airflow_version)
```

Comment:
We should also loop through the `for (provider_id, provider_versions) in provider_metadata.items()` here - otherwisee provider_version will be None and it will always choose the latest version (we could also pass `provider_version` as parameter instead (incluing "latest") if we want to just build one provider version sbom,

Reply:
Done.

We can now give `provider-id` and `provider-version`.

`--provider-id`:
- specific id -> 1 provider
- None -> all provider ids found in the metadata file

`--provider-version`:
- fixed version only allowed if `--provider-id` is a specific provider (not None)
- 'latest' will pull the latest version for each selected providers.
- None will use all the historical versions.